packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -21,11 +35,45 @@ export interface AccordionProps extends Pick<AccordionContainerProps, "class" |
 export function Accordion(props: AccordionProps): ReactElement | null {
     const reducer = useRef(getCollapsedAccordionGroupsReducer(props.singleExpandedGroup ? "single" : "multiple")); // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
 
-    const [collapsedAccordionGroups, collapsedAccordionGroupsDispatch] = useReducer(
+    const [accordionGroupCollapsedState, accordionGroupCollapsedStateDispatch] = useReducer(
         reducer.current,
-        props.groups.map(() => !props.previewMode && props.collapsible)
+        undefined,
+        () => {
+            const groupCollapsedStates = props.groups.map(
+                group => !props.previewMode && props.collapsible && !!group.initiallyCollapsed
+            );
+
+            if (!props.previewMode && props.singleExpandedGroup) {
+                const lastGroupCollapsedStateIndex = groupCollapsedStates.lastIndexOf(false);
+
+                if (lastGroupCollapsedStateIndex > -1) {
+                    for (let i = 0; i < lastGroupCollapsedStateIndex; i++) {
+                        groupCollapsedStates[i] = true;
```

Comment:
I don't quite get what the purpose of this is. In the initialization of the reducer (or resetting), we say that if we are not in previewmode and the feature to only have a single expanded group is on during runtime we check for the last group that is not collapsed. If there is a group that is not collapsed, then we set everything before to collapsed.

But isn't this technically handled by the consistency check that you added?

Reply:
It's partially handled by the consistency check. The 'Collapsed' and 'Start as collapsed' properties are dynamic which cannot be evaluated during modelling time. Therefore, we have this builtin mechanism to keep the widget adhere to its configuration.

Reply:
Ahhh yeah I forgot about the dynamic values, bummer 😢